drivers/bm/bm_dfu_client.c

```diff
@@ -0,0 +1,439 @@
+#include <init.h>
+#include <logging/log.h>
+#include <storage/flash_map.h>
+#include <bootutil/bootutil_public.h>
+#include <dfu/mcuboot.h>
+#include <sys/reboot.h>
+
+#include <drivers/bm/bm_serial.h>
+#include <drivers/bm/bm_dfu.h>
+#include <drivers/bm/bm_dfu_client.h>
+#include <drivers/bm/bm_common.h>
+
+LOG_MODULE_REGISTER(bm_dfu_client, CONFIG_BM_LOG_LEVEL);
+
+static dfu_client_ctx_t _client_context;
+
+static void bm_dfu_client_req_next_chunk(void)
+{
+    bm_frame_header_t frm_hdr;
+    bm_frame_t *chunk_req_frm;
+    bm_dfu_event_chunk_request_t chunk_req_evt;
+    uint8_t tx_buf[sizeof(bm_frame_header_t) + sizeof(bm_dfu_event_chunk_request_t) + sizeof(bm_dfu_frame_header_t)];
+
+    /* Stuff BM Frame Header*/
+    frm_hdr.version = BM_V0;
+    frm_hdr.payload_type = BM_DFU;
+    frm_hdr.payload_length = sizeof(bm_dfu_event_chunk_request_t) + sizeof(bm_dfu_frame_header_t);
```

Comment:
Looks like this is repeated a few times. I'd recommend making it a function (even if just an inline one) in case we ever have to modify it (so we don't have to search everywhere to see where we need the changes made)

Reply:
(with length as the argument)